frontend/src/metabase/api/dataset.ts

```diff
@@ -1,9 +1,23 @@
-import type { NativeQueryForm, DatasetQuery } from "metabase-types/api";
+import type {
+  NativeQueryForm,
+  DatasetQuery,
+  CardQueryMetadata,
+} from "metabase-types/api";
 
 import { Api } from "./api";
+import { provideAdhocQueryMetadataTags } from "./tags";
 
 export const datasetApi = Api.injectEndpoints({
   endpoints: builder => ({
+    getAdhocQueryMetadata: builder.query<CardQueryMetadata, DatasetQuery>({
```

Comment:
`CardQueryMetadata` uses `Table` type which is currently inaccurate for card-based _tables_.
We might want to improve the typing there in the near future.

![image](https://github.com/metabase/metabase/assets/6830683/6215c929-e05a-4fcd-9cff-30d703bbb524)